index.js

```diff
@@ -197,26 +214,90 @@ function uploadMergeFiles(viewId, mergeData, templatesToRecords){
         return Promise.all(uploadPromises);
     })
     .then((uploadResponses) =>{
-        console.log("done uploading everything");
+        log.log("done uploading everything");
+
         if(globalCallback){
             globalCallback(null, "Merge completed successfully");
         }
     })
     .catch(function(err) {
-        handleError(err);
+      checkFieldNames("merged", config.merged_doc_table.view_id);
+      handleError(err);
     });
 }
 
 /**
+ * Function to determine what field is causing the upload error
+ * @param {String} table
+ * @param {Number} viewId
+ */
+function checkFieldNames(table, viewId) {
```

Comment:
@jetherton This should probably be refactored -> 1. There are too many loops over the fields 2. It is too verbose. It'd be great if we could discuss some refactoring that could be applied when you get a chance.